### PR TITLE
Fix Arrow export for nested records

### PR DESCRIPTION
**Pull request by dominiklohmann**
###  :notebook_with_decorative_cover: Description

This fixes a small oversight from the recent unflattening PR.

###  :memo: Checklist

- [x] All user-facing changes have changelog entries.
- [x] The changes are reflected on [docs.tenzir.com/vast](https://docs.tenzir.com/vast), if necessary.
- [x] The PR description contains instructions for the reviewer, if necessary.

### :dart: Review Instructions

Should be easy to verify locally like this:

```
❯ vast -N import suricata -r vast/integration/data/suricata/eve.json
❯ vast -N export arrow '"" in #type'
```

### CODE REVIEW

File: libvast/src/format/arrow.cpp [hunk 1]
```diff
@@ -15,15 +15,14 @@
 
 #if VAST_ENABLE_ARROW
 
-#  include "vast/format/arrow.hpp"
-
```
Comment (tobim) [inline libvast/src/format/arrow.cpp:19]: Didn't we once have an `IncludeCategory` for this?

File: libvast/src/format/arrow.cpp [hunk 2]
```diff
@@ -50,14 +49,16 @@ writer::~writer() {
 
 caf::error writer::write(const table_slice& slice) {
   if (out_ == nullptr)
-    return ec::filesystem_error;
+    return caf::make_error(ec::filesystem_error, "invalid arrow output stream");
```
Comment (tobim) [inline libvast/src/format/arrow.cpp:52]:
```suggestion
    return make_error(ec::filesystem_error, "invalid arrow output stream");
```

File: libvast/src/format/arrow.cpp [hunk 3]
```diff
@@ -50,14 +49,16 @@ writer::~writer() {
...
   if (!layout(slice.layout()))
-    return ec::unspecified;
+    return caf::make_error(ec::logic_error, "failed to update layout");
```
Comment (tobim) [inline libvast/src/format/arrow.cpp:54]:
```suggestion
    return make_error(ec::logic_error, "failed to update layout");
```

File: libvast/src/format/arrow.cpp [hunk 4]
```diff
@@ -50,14 +49,16 @@ writer::~writer() {
...
   // Get the Record Batch and print it.
   auto batch = as_record_batch(slice);
   VAST_ASSERT(batch != nullptr);
-  if (!current_batch_writer_->WriteRecordBatch(*batch).ok())
-    return ec::filesystem_error;
+  if (auto status = current_batch_writer_->WriteRecordBatch(*batch);
+      !status.ok())
+    return caf::make_error(ec::filesystem_error, "failed to write record batch",
```
Comment (tobim) [inline libvast/src/format/arrow.cpp:60]:
```suggestion
    return make_error(ec::filesystem_error, "failed to write record batch",
```

File: libvast/src/format/arrow.cpp (hunk 2 above)
Comment (tobim) [reply libvast/src/format/arrow.cpp:52]: Isn't this actually a logic error?

File: libvast/src/format/arrow.cpp (hunk 4 above)
Comment (tobim) [reply libvast/src/format/arrow.cpp:60]: We should introduce an `arrow_error` for these.

**Review by tobim — CHANGES_REQUESTED** [review]

File: libvast/src/format/arrow.cpp (hunk 2 above)
Comment (dominiklohmann) [reply libvast/src/format/arrow.cpp:52]: Yes, makes sense to change.

File: libvast/src/format/arrow.cpp (hunk 4 above)
Comment (dominiklohmann) [reply libvast/src/format/arrow.cpp:60]: I'd rather make this `ec::unspecified`, as the Arrow Status contains its own category. In the case of this error, it was `Invalid`.

File: libvast/src/format/arrow.cpp (hunk 1 above)
Comment (dominiklohmann) [reply libvast/src/format/arrow.cpp:19]: We still do. I think this is a change in recent clang-format versions. Let's just leave it as formatted by clang-format here.

File: libvast/src/format/arrow.cpp (hunk 3 above)
Comment (dominiklohmann) [reply libvast/src/format/arrow.cpp:54]: Not doing these. `vast::make_error` is just an alias for `caf::make_error`, and we use the latter more than the former in the code base.

File: libvast/src/format/arrow.cpp (hunk 4 above)
Comment (tobim) [reply libvast/src/format/arrow.cpp]: That's ok as well, the inner error is what matters.

File: libvast/src/format/arrow.cpp (hunk 3 above)
Comment (tobim) [reply libvast/src/format/arrow.cpp:54]: Just for the record:
```
$ git grep 'make_error(' | wc -l
343
$ git grep 'caf::make_error(' | wc -l
24
```

But it seems I have been outvoted on this issue.

**Review by tobim — APPROVED** [review]

File: libvast/src/format/arrow.cpp (hunk 3 above)
Comment (dominiklohmann) [reply libvast/src/format/arrow.cpp:54]: It's not as easy to grep for just error, but for `caf::error` the numbers were the other way around. ;)

**Pull request merged**